src/asmcnc/apps/maintenance_app/widget_maintenance_spindle_settings.py

```diff
@@ -377,49 +374,71 @@ def autofill_rpm_time(self):
-    #             Clock.schedule_once(self.get_spindle_info, 0.3)
-    #         else:
-    #             popup_info.PopupError(self.sm, self.l, self.l.get_str("Please ensure machine is idle before continuing."))
-    #     else:
-    #         self.uptime_label.text = "Uptime: " + str(int(self.m.spindle_brush_use_seconds/3600)) + " hours"
-
-    # def get_spindle_info(self, dt):
-    #     self.m.s.write_protocol(self.m.p.GetDigitalSpindleInfo(), "GET DIGITAL SPINDLE INFO")
-    #     self.check_info_count = 0
-    #     Clock.schedule_once(self.check_spindle_info, 0.3)
-
-    # def check_spindle_info(self, dt):
-    #     self.check_info_count += 1
-    #     # Value of -999 represents disconnected spindle
-    #     if (self.m.s.digital_spindle_ld_qdA != -999 and self.m.s.spindle_serial_number not in [None, -999, 999]) or (self.check_info_count > 10):
-    #         self.read_restore_info()
-    #     else: # Keep trying for a few seconds
-    #         Clock.schedule_once(self.check_spindle_info, 0.3)
-
-    # def read_restore_info(self):
-    #     self.m.s.write_command('M5')
-    #     self.wait_popup.popup.dismiss()
-    #     # Value of -999 for ld_qdA represents disconnected spindle
-    #     if self.m.s.digital_spindle_ld_qdA != -999 and self.m.s.spindle_serial_number not in [None, -999, 999]:
-    #         # Get info was successful, show info
-    #         self.uptime_label.text = "Uptime: " + str(int(self.m.s.spindle_brush_run_time_seconds/3600)) + " hours"
-    #     else:
-    #         # Otherwise, spindle is probably disconnected
-    #         error_message = self.l.get_str("No SC2 Spindle motor detected.") + " " + self.l.get_str("Please check your connections.")
-    #         popup_info.PopupError(self.sm, self.l, error_message)
+        self.rpm_override = False
+
+    def hide_spindle_data_container(self):
+        self.spindle_data_container.size_hint_x = 0
+        self.spindle_data_container.opacity = 0
+        self.spindle_data_container.parent.spacing = 0
+        self.get_data_button.disabled = True
+
+    def show_spindle_data_container(self):
+        self.spindle_data_container.size_hint_x = 0.75
+        self.spindle_data_container.opacity = 1
+        self.spindle_data_container.parent.spacing = 15
+        self.get_data_button.disabled = False
+
+    def show_spindle_data_popup(self):
+        popup_maintenance.PopupGetSpindleData(self.sm, self.l)
+
+    def raise_z_then_get_data(self):
+        if self.m.state().startswith('Idle'):
+            self.wait_popup = popup_info.PopupWait(self.sm, self.l, self.l.get_str('SmartBench is raising the Z axis.'))
+            self.m.zUp()
+            Clock.schedule_once(self.get_spindle_data, 0.4)
+        else:
+            popup_info.PopupError(self.sm, self.l, self.l.get_str("Please ensure machine is idle before continuing."))
+
+    def get_spindle_data(self, dt):
+        # Wait until Z axis is raised
+        if self.m.state().startswith('Idle'):
```

Comment:
probably safer to use if not self.m.smartbench_is_busy() instead - it's a bit more thorough